block-producer/src/txqueue/mod.rs

```diff
@@ -175,17 +177,22 @@ where
             let ready_queue = self.ready_queue.clone();
             let batch_builder = self.batch_builder.clone();
 
-            tokio::spawn(async move {
-                match batch_builder.build_batch(txs.clone()).await {
-                    Ok(_) => {
-                        // batch was successfully built, do nothing
-                    },
-                    Err(_) => {
-                        // batch building failed, add txs back at the end of the queue
-                        ready_queue.write().await.append(&mut txs);
-                    },
+            tokio::spawn(
+                async move {
+                    match batch_builder.build_batch(txs.clone()).await {
+                        Ok(_) => {
+                            // batch was successfully built, do nothing
+                        },
+                        Err(_) => {
+                            // batch building failed, add txs back at the end of the queue
+                            ready_queue.write().await.append(&mut txs);
+                        },
+                    }
                 }
-            });
+                .instrument(
+                    info_span!(target: COMPONENT, "block_producer::batch_builder::build_batch"),
+                ),
```

Comment:
I would probably remove this as well.

Reply:
@bobbinth I tried to remove this and then it starts to log children as root spans. I'll keep it for now, just reduce to `block_producer`, but need to investigate its behavior more.

Reply:
In this case, I'd probably use `batch_builder` rather than `block_producer`.

Does this span get closed properly? (or is it just a few lines missing in the logs you uploaded?)

Reply:
@bobbinth thank you! I've fixed it. Yes, I missed 2 lines of logs, now it should be good.